torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1511,12 +1561,25 @@ def _verify(expr, potential_expr):
             # user directives about relationships, we can remove this check from
             # verification.
             if len(expr.free_symbols) == 1:
-                srcs = symbol_to_source[expr.free_symbols.pop()]
+                symbol = expr.free_symbols.pop()
+                # NOTE! Manual user directives override the rule around not allowing any constraining of dynamic dims
+                # [NOTE - on user_constrained]
+                # User constrained symbols have an entry in var_to_range, but some of their range
+                # comes from manual user directives like dynamo's constrain api. This distinction is maintained
+                # for the purposes of protecting and enforcing user directives. In strict_mark_dyn mode, we do not
+                # allow ANY constraining of values for a given symbol. However, if this symbol is user constrained, we
+                # relax this requirement in favor of honoring the user directive and allowing the user specified range.
+                # We do this by checking of the range is unconstrained*.
+                #
+                # *unconstrained might mean a default unsound 0/1 constraint.
+                if self.var_to_range[symbol] != _default_value_range(specialize_zero_one=self.specialize_zero_one):
+                    return
+                srcs = symbol_to_source[symbol]
                 for src in srcs:
                     if src in dynamic_sources:
-                        raise RuntimeError(f"Attempting to introduce a guard {potential_expr} that violates user's mark_dynamic")
+                        raise RuntimeError(f"Attempting to introduce a guard {potential_expr} that violates user's constraint")
```

Comment:
So for a symbol that is known to be equal to (replacing) another symbol, both their sources will be in `srcs`, yeah? It will be cleaner / more informative to collect all the sources together and error once reporting all of them. This would indicate not only that, e.g., `s0 < 512` but also `s0` was discovered equal to `s1`. Otherwise what if `s1`'s source is in `dynamic_sources` and not `s0`'s...confusing!

Reply:
Sure, we can improve that UX in a followup PR. I think it's kind of a distraction here.

Reply:
sg